### PR TITLE
[RHELC-878] Remove 8.4 from supported EUS versions

### PR DESCRIPTION
**Pull request by pr-watson**
This PR removes the CentOS 8.4 EUS repos and their links to integration tests

Jira Issues: [RHELC-878](https://issues.redhat.com/browse/RHELC-878)

Checklist

- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [x] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Comment by pr-watson** [comment]
@danmyway since this PR removes RHEL 8.4 repos I believe the integration test for centos-8.4 need to be updated to use centos-8.6.

**Comment by r0x0d** [comment]
@pr-watson, I think you can drop the 8.4 versions from packit too. @danmyway, do you agree?

**Comment by danmyway** [comment]
> @pr-watson, I think you can drop the 8.4 versions from packit too. @danmyway, do you agree?

I agree with dropping it from the config.
However, I have a question about a timeline for this PR - will this get merged ASAP? 
Are we dropping the upstream verifications for 8.4 right now?
I suppose we won't do regression testing for the `1.3` given the RHEL 8.4 goes of out of Extended Update Support (EUS) by the time of the release.
But IMO keeping the 8.4 verifications live until May, 31st seems reasonable to me.
Again, remove it from the config in this PR at your will, I am not opposed to that at all.
Thoughts @SpyTec @bocekm ?

**Comment by pr-watson** [comment]
> > @pr-watson, I think you can drop the 8.4 versions from packit too. @danmyway, do you agree?
> 
> I agree with dropping it from the config. However, I have a question about a timeline for this PR - will this get merged ASAP? Are we dropping the upstream verifications for 8.4 right now? I suppose we won't do regression testing for the `1.3` given the RHEL 8.4 goes of out of Extended Update Support (EUS) by the time of the release. But IMO keeping the 8.4 verifications live until May, 31st seems reasonable to me. Again, remove it from the config in this PR at your will, I am not opposed to that at all. Thoughts @SpyTec @bocekm ?

My PR for removing the 8.4 repos from Dogfood servers has been postponed for merging till May 31, 2023, so I assume the same would apply for this PR.

**Comment by SpyTec** [comment]
For dogfood servers it makes sense as we want to be able to test against it until last second. However, I don't think we need to block this pr. We have the release branch if we need to release something urgently 

Packit is also related to testing so I'd want to keep that for now, at least until we drop support for 8.4. But I assume that would cause issues for all PRs right?

**Comment by danmyway** [comment]
> For dogfood servers it makes sense as we want to be able to test against it until last second. However, I don't think we need to block this pr. We have the release branch if we need to release 
something urgently 
> 
> Packit is also related to testing so I'd want to keep that for now, at least until we drop support for 8.4. But I assume that would cause issues for all PRs right? 

If we would merge this PR, we need to remove the 8.4 from Packit config, otherwise most of the CentOS 8.4 pipeline will fail as you say @SpyTec.
So in case we want to drop the upstream verifications now, let me know and I will also remove all the 8.4 related code from tests.

**Comment by SpyTec** [comment]
> If we would merge this PR, we need to remove the 8.4 from Packit config, otherwise most of the CentOS 8.4 pipeline will fail as you say 

That's what I feared. Then we should wait with this PR too and do Packit then

**Comment by danmyway** [comment]
> That's what I feared. Then we should wait with this PR too and do Packit together with this PR

Cool, thank you for the clarification.
I can take care of it @pr-watson, and I will also remove the 8.4 related test code.

**Comment by r0x0d** [comment]
There is a PR that removes the 8.4 support from our internal vagrant workflow. The PR will be merged once this one is merged, so, this is a future warning for myself.

File: tests/integration/tier1/removed-pkgs-centos-85/ansible/install_removed_pkgs_from_centos_85_repos.yml
Comment (danmyway) [inline tests/integration/tier1/removed-pkgs-centos-85/ansible/install_removed_pkgs_from_centos_85_repos.yml:1]: Hey, @pr-watson, I will revert changes in this file.
We need to add and enable the 8.4 repos, because we are looking for a package, which is not present in 8.5 repos anymore.

File: plans/tier1.fmf
Comment (danmyway) [inline plans/tier1.fmf:31]: There's no CentOS 8.6

Comment (danmyway) [reply plans/tier1.fmf:31]: I will disable the plan until we have 8.6 in pipeline

**Comment by danmyway** [comment]
I have rebased and modified the tests @pr-watson

File: tests/integration/tier0/latest-kernel-check/test_latest_kernel_check.py
Comment (kokesak) [inline tests/integration/tier0/latest-kernel-check/test_latest_kernel_check.py:17]: the 8.4 are removed no?
```suggestion
    # For CentOS, we are working with hardcoded repos in /usr/share/convert2rhel/repos/centos-8.5
```

**Review by kokesak — COMMENTED** [review]
Just one small thing in the comment. Otherwise, the QE side looks good!

File: tests/integration/tier0/latest-kernel-check/test_latest_kernel_check.py
Comment (danmyway) [reply tests/integration/tier0/latest-kernel-check/test_latest_kernel_check.py:17]: Thanks for spotting this

File: convert2rhel/unit_tests/pkgmanager/handlers/dnf/dnf_test.py
Comment (r0x0d) [inline convert2rhel/unit_tests/pkgmanager/handlers/dnf/dnf_test.py:73]:
```suggestion
        assert instance._base.conf.substitutions["releasever"] == "8.5"
```

File: convert2rhel/unit_tests/systeminfo_test.py
Comment (r0x0d) [inline convert2rhel/unit_tests/systeminfo_test.py:445]:
```suggestion
    assert "8.5" in caplog.records[-3].message
```

File: tests/ansible_collections/roles/add-custom-repos/main.yml
Comment (r0x0d) [inline tests/ansible_collections/roles/add-custom-repos/main.yml:12]: I think this part should include the 8.6 instead of just removing the rhel8-eus-repos, right? 

```yaml
- import_playbook: rhel8-eus-repos.yml
  when: ansible_facts['distribution_version'] == "8.6"
```

Since 8.6 will be our lowest minor version supported for EUS, we should enable it for that version, look at this comment: https://github.com/oamg/convert2rhel/pull/808/files#diff-f3b2429fb7abe33e7f90d47b80809bf722ae4459862b257f57138761a4d03140R8

@danmyway my reasoning here is correct?

**Comment by pr-watson** [comment]
@r0x0d thanks for the suggestions

File: tests/ansible_collections/roles/add-custom-repos/main.yml
Comment (danmyway) [reply tests/ansible_collections/roles/add-custom-repos/main.yml:12]: You're right @r0x0d, I will put it back

Comment (danmyway) [reply tests/ansible_collections/roles/add-custom-repos/main.yml:12]: Thanks for noticing

**Comment by danmyway** [comment]
Rebased, conflicts resolved, ready to merge

**Review by r0x0d — DISMISSED** [review]

File: convert2rhel/unit_tests/subscription_test.py
Comment (r0x0d) [inline convert2rhel/unit_tests/subscription_test.py:1412]: I think you could also do something like: 

```suggestion
    eus_version = EUS_MINOR_VERSION[0].split(".")
    major_version = eus_version[0]
    minor_version = eus_version[1]
    monkeypatch.setattr(system_info, "version", namedtuple("Version", ["major", "minor"])(major_version, minor_version))
```

Would look a bit cleaner and easier to understand. No need to change right now, just writing down the suggestion.

**Review by r0x0d — APPROVED** [review]
Looks good!

File: convert2rhel/unit_tests/subscription_test.py
Comment (pr-watson) [reply convert2rhel/unit_tests/subscription_test.py:1412]: It does look better! I can add that in, wondering if we should wait till the first version is merged or not.

Comment (r0x0d) [reply convert2rhel/unit_tests/subscription_test.py:1412]: Since everything passed, I would say to merge this first and then later do this change. No worries.

**Pull request merged**
